### PR TITLE
[fix] Fix llama4 + eagle3

### PR DESCRIPTION
**Pull request by mikeiovine**
# [fix] Fix llama4 + eagle3

## Description

A few problems here due to https://github.com/NVIDIA/TensorRT-LLM/pull/3383.
* The new `Llama4ForConditionalGeneration` subclass was not passing on the `return_context_logits` to the base class properly. It was being passed as a positional argument but the position was off by one. The result is that the parent class would see `pipeline_interface = True` (a type error) and `return_context_logits = False` (the default), causing torch to crash when the logits are gathered in `ModelEngine`.
* The new subclass also was not passing on the `spec_metadata` at all.
* (Unrelated to the PR linked above) The draft and target model engines need to have the same max seq len.

## Test Coverage

We already have Eagle 3 tests, but not for llama 4. These need to be added in follow ups. It is not trivial to enable here since we will have to add the latest llama 4 eagle checkpoints to our model repository + make sure the new tests won't time out (we already have a ton of very slow llama 4 tests).

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

**Comment by hlu1** [comment]
A type checker would have caught this error.

**Review by hlu1 — APPROVED** [review]

**Review by lfr-0531 — APPROVED** [review]
